webapp/src/components/boardTemplateSelector/boardTemplateSelector.tsx

```diff
@@ -0,0 +1,168 @@
+import CloseIcon from '../../widgets/icons/close'
+import AddIcon from '../../widgets/icons/add'
+import Button from '../../widgets/buttons/button'
+import octoClient from '../../octoClient'
+import mutator from '../../mutator'
+import {getTemplates, getCurrentBoard} from '../../store/boards'
+import {fetchGlobalTemplates, getGlobalTemplates} from '../../store/globalTemplates'
+import {useAppDispatch, useAppSelector} from '../../store/hooks'
+import TelemetryClient, {TelemetryActions, TelemetryCategory} from '../../telemetry/telemetryClient'
+
+import BoardTemplateSelectorPreview from './boardTemplateSelectorPreview'
+import BoardTemplateSelectorItem from './boardTemplateSelectorItem'
+
+import './boardTemplateSelector.scss'
+
+type Props = {
+    title?: React.ReactNode
+    description?: React.ReactNode
+    onClose?: () => void
+}
+
+const BoardTemplateSelector = React.memo((props: Props) => {
+    const globalTemplates = useAppSelector<Board[]>(getGlobalTemplates) || []
+    const currentBoard = useAppSelector<Board>(getCurrentBoard) || null
+    const {title, description, onClose} = props
+    const dispatch = useAppDispatch()
+    const intl = useIntl()
+    const history = useHistory()
+    const match = useRouteMatch<{boardId: string, viewId?: string}>()
+
+    const showBoard = useCallback(async (boardId) => {
+        const params = {...match.params, boardId: boardId || ''}
+        delete params.viewId
+        const newPath = generatePath(match.path, params)
+        history.push(newPath)
+        if (onClose) {
+            onClose()
+        }
+    }, [match, history, onClose])
+
+    useEffect(() => {
+        if (octoClient.workspaceId !== '0' && globalTemplates.length === 0) {
+            dispatch(fetchGlobalTemplates())
+        }
+    }, [octoClient.workspaceId])
+
+    const onBoardTemplateDelete = useCallback((template: Board) => {
+        TelemetryClient.trackEvent(TelemetryCategory, TelemetryActions.DeleteBoardTemplate, {board: template.id})
+        mutator.deleteBlock(
+            template,
+            intl.formatMessage({id: 'BoardTemplateSelector.delete-template', defaultMessage: 'Delete template'}),
+            async () => {
+            },
+            async () => {
+                showBoard(template.id)
+            },
+        )
+    }, [showBoard])
+
+    const unsortedTemplates = useAppSelector(getTemplates)
+    const templates = useMemo(() => Object.values(unsortedTemplates).sort((a: Board, b: Board) => a.createAt - b.createAt), [unsortedTemplates])
+    const allTemplates = globalTemplates.concat(templates)
+
+    const [activeTemplate, setActiveTemplate] = useState<Board>(allTemplates[0])
+
+    useEffect(() => {
+        if (!activeTemplate) {
+            setActiveTemplate(templates.concat(globalTemplates)[0])
+        }
+    }, [templates, globalTemplates])
+
+    if (!allTemplates) {
+        return <div/>
+    }
+
+    return (
+        <div className='BoardTemplateSelector'>
```

Comment:
What convention have we adopted for css class names?  We have camel-case and lowercase with hyphens mixed.  Is there some rule that determines which to use?

Reply:
In this specific case is because the internal convention is to use the component name as the class of the root element of the component. This way, any where in the css you can use the class `.BoardTemplateSelector` to affect the `BoardTemplateSelector` component, and also, the `boardTemplateSelector.scss` is going to start with the `.BoardTemplateSelector` class as the root of the scss to scope the styles to only affect the current component.